src/lib/Menu/MainMenuBuilder.php

```diff
@@ -87,15 +84,19 @@ public function createStructure(array $options): ItemInterface
         $menu->addChild($this->factory->createItem(self::ITEM_CONTENT, []));
         $menu[self::ITEM_CONTENT]->setChildren($contentMenuItems);
 
-        $adminMenuItems = $this->getAdminMenuItems();
-        $menu->addChild($this->factory->createItem(self::ITEM_ADMIN, []));
-        $menu[self::ITEM_ADMIN]->setChildren($adminMenuItems);
+        if ($this->permissionResolver->hasAccess('setup', 'administrate')) {
+            $adminMenuItems = $this->getAdminMenuItems();
+            $menu->addChild($this->factory->createItem(self::ITEM_ADMIN, []));
```

Comment:
It's a breaking change for custom menu listeners adding items under `ITEM_ADMIN`.